python/remotivelabs-broker/remotivelabs/broker/sync/client.py

```diff
@@ -159,69 +166,38 @@ def connect(self, url: str, api_key: Union[str, None] = None):
         if self.on_connect is not None:
             self.on_connect(self)
 
-    def _validate_and_get_subscribed_signals(
-        self, subscribed_namespaces: List[str], subscribed_signals: List[str]
-    ) -> List[SignalIdentifier]:
-        # Since we cannot know which List[signals] belongs to which namespace we need to fetch
-        # all signals from the broker and find the proper signal with namespace. Finally,  we
-        # also filter out namespaces that we do not need since we might have duplicated signal names
-        # over namespaces
-        # Begin
-
-        def verify_namespace(available_signal: SignalIdentifier):
-            return list(
-                filter(
-                    lambda namespace: available_signal.namespace == namespace,
-                    subscribed_namespaces,
-                )
-            )
-
-        def find_subscribed_signal(available_signal: SignalIdentifier):
-            return list(filter(lambda s: available_signal.name == s, subscribed_signals))
-
-        available_signals: List[SignalIdentifier] = list(filter(verify_namespace, self.list_signal_names()))
-        signals_to_subscribe_to: List[SignalIdentifier] = list(filter(find_subscribed_signal, available_signals))
-
-        # Check if subscription is done on signal that is not in any of these namespaces
-        signals_subscribed_to_but_does_not_exist = set(subscribed_signals) - set(map(lambda s: s.name, signals_to_subscribe_to))
-
-        if len(signals_subscribed_to_but_does_not_exist) > 0:
-            raise BrokerException(f"One or more signals you subscribed to does not exist " f", {signals_subscribed_to_but_does_not_exist}")
-
-        return list(map(lambda s: SignalIdentifier(s.name, s.namespace), signals_to_subscribe_to))
-
     def subscribe(
         self,
-        signal_names: List[str],
-        namespaces: List[str],
+        signals_to_subscribe_to: Union[List[SignalIdentifier], List[str]],
         on_signals: Optional[Callable[[SignalsInFrame], None]] = None,
         changed_values_only: bool = True,
     ):
+        client_id = br.common_pb2.ClientId(id="subscribe-sample")
         if on_signals is None and self.on_signals is None:
             raise BrokerException(
                 "You have not specified global client.on_signals nor client.subscribe(on_signals=callback), "
                 "or you are invoking subscribe() before client.on_signals which is not allowed"
             )
 
-        client_id = br.common_pb2.ClientId(id=self.client_id)
-        signals_to_subscribe_to: List[SignalIdentifier] = self._validate_and_get_subscribed_signals(namespaces, signal_names)
-
-        def to_protobuf_signal(s: SignalIdentifier):
+        def to_protobuf_signal(s: Union[SignalIdentifier, str]):
```

Comment:
Should we add leading `_` since this should be internal?

Reply:
Since its inside a function I do not think it matters, right? But I am not sure here either...

Reply:
I missed that part. Looks good